zip-plus/src/code/iprs/pntt/radix8/octet_reversal.rs

```diff
@@ -0,0 +1,64 @@
+/// Reverse octets (triples of consecutive binary digits)
+/// in a given number `x`. The total number of octets
+/// constituting the number is also provided as the argument
+/// `num_octets`.
+#[allow(clippy::arithmetic_side_effects)]
+pub(crate) fn octet_reversal(x: usize, num_octets: usize) -> usize {
+    assert_ne!(num_octets, 0, "The number of octets cannot be zero");
+
+    let mut result = 0;
+
+    let mut left_octet_mask = 0b111;
+    let mut right_octet_mask = 0b111 << (3 * (num_octets - 1));
+
+    for i in 0..(num_octets / 2 + num_octets % 2) {
```

Comment:
```suggestion
    for i in 0..num_octets >> 1 {
```

If you care about 🏎️  speed 🏎️  :P 

The inner (if num_octets is odd) octet doesn't have to be swapped with itself :)

Reply:
Good point!
Note though that `num_octets >> 1` is a terrible code compared to `num_octets / 2`, and they both compile [exactly the same](https://godbolt.org/z/rzvMKqhx8)

Reply:
Turns out, we do need `+ num_octets % 2`, otherwise the mask bit shift doesn't do the right thing

Reply:
¯\_(ツ)_/¯ interesting